dev/reference/alert-rules.md

```diff
@@ -0,0 +1,1276 @@
+
+    `sum(tidb_ddl_waiting_jobs) > 5`
+
+* Description:
+
+    When the number of DDL tasks pending for execution in TiDB exceeds 5, an alert is triggered.
+
+* Solution:
+
+    Check whether there is any time-consuming `add index` operation that is being executed by running `admin show ddl`.
+
+## PD alert rules
+
+This section gives the alert rules for the PD component.
+
+### Emergency-level alerts
+
+Emergency-level alerts are often caused by a service or node failure. Manual intervention is required immediately.
+
+#### `PD_cluster_offline_tikv_nums`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_down_count"}) > 0`
+
+* Description:
+
+    PD has not received a TiKV heartbeat for a long time (the default configuration is 30 minutes).
+
+* Solution:
+
+    * Check whether the TiKV process is normal, the network is isolated or the load is too high, and recover the service as much as possible.
+    * If the TiKV cannot be recovered, you can make it offline.
+
+### Critical-level alerts
+
+For the critical-level alerts, a close watch on the abnormal metrics is required.
+
+#### `PD_etcd_write_disk_latency`
+
+* Alert rule:
+
+    `histogram_quantile(0.99, sum(rate(etcd_disk_wal_fsync_duration_seconds_bucket[1m])) by (instance,job,le)) > 1`
+
+* Description:
+
+    etcd writes data to disk with a slower speed than normal. It might lead to PD leader timeout or failure to store TSO on disk in time, which will shut down the service of the entire cluster.
+
+* Solution:
+
+    * Find the cause of slow writes. It might be other services that overload the system. You can check whether PD itself occupies a large amount of CPU or IO resources.
+    * Try to restart PD or manually transfer leader to another PD to recover the service.
+    * Make offline and replace the PD with issues if it cannot be recovered due to environmental factors.
+
+#### `PD_miss_peer_region_count`
+
+* Alert rule:
+
+    `sum(pd_regions_status{type="miss_peer_region_count"}) > 100`
+
+* Description:
+
+    The number of Region replicas is smaller than the value of `max-replicas`. When the TiKV is down and its downtime exceeds `max-down-time`, it usually leads to missing replicas for some Regions during a period of time. When a TiKV node is made offline, it might result in a small number of Regions with missing replicas.
+
+* Solution:
+
+    * Find the cause of the issue by checking whether there is any TiKV that is down or being made offline.
+    * Watch the region health panel and see whether `miss_peer_region_count` is continuously decreasing.
+
+### Warning-level alerts
+
+Warning-level alerts are a reminder for an issue or error.
+
+#### `PD_cluster_lost_connect_tikv_nums`
+
+* Alert rule:
+
+    `sum(pd_cluster_status{type="store_disconnected_count"}) > 0`
+
+* Description:
+
+    PD does not receive a TiKV heartbeat within 20 seconds. Normally a TiKV heartbeat comes in every 10 seconds.
+
+* Solution:
+
+    * Check whether the TiKV is being restarted.
```

Comment:
`the TiKV` is not a complete phrase. We can use `the TiKV instance/node` or just the lone `TiKV`, but not `the TiKV`. Please check and update all similar occurrences in this PR. Thanks!